src/aap_eda/services/activation/engine/ports.py

```
@@ -19,54 +19,44 @@ def render_string(value: str, context: dict) -> str:
 
 
 def find_ports(rulebook_text: str, context: dict = None) -> list[tuple]:
```

Ruff (0.15.18)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)